src/tests/suite/daos_test_common.c

```diff
@@ -1026,3 +1026,147 @@ get_daos_prop_with_user_acl_perms(uint64_t perms)
 	D_FREE(user);
 	return prop;
 }
+
+int
+get_server_config(char *host, char *server_config_file)
+{
+	char	command[256];
+	size_t	len = 0;
+	size_t	read;
+	char	*line = NULL;
+	char	*pch;
+
+	snprintf(command, sizeof(command),
+		 "ssh %s ps ux -A | grep daos_server | grep start", host);
+	FILE *fp = popen(command, "r");
+
+	print_message("Command %s\n", command);
+	if (fp == NULL)
+		return -DER_INVAL;
+
+	while ((read = getline(&line, &len, fp)) != -1) {
+		print_message("line %s\n", line);
+		if (strstr(line, "--config") != NULL ||
+		    strstr(line, "-o") != NULL)
+			break;
+	}
+
+	pch = strtok(line, " ");
+	while (pch != NULL) {
+		if (strstr(pch, "yaml") != NULL &&
+		    strstr(pch, "--config") != NULL) {
+			strcpy(server_config_file, strchr(pch, '=') + 1);
+			break;
+		}
+
+		if (strstr(pch, "yaml") != NULL) {
+			strcpy(server_config_file, pch);
+			break;
+		}
+		pch = strtok(NULL, " ");
+	}
```

Comment:
this is not working, instead I would suggest just coping everything after -o or --config returned after ps aux command.

Reply:
Updating the code to pick the next string after -o and --config